src/test/java/com/redhat/devtools/lsp4ij/fixtures/LSPCodeActionFixtureTestCase.java

```diff
@@ -0,0 +1,150 @@
+     */
+    protected void assertCodeActions(@NotNull String fileName,
+                                     @NotNull String editorContentText,
+                                     @NotNull String codeActionsJson,
+                                     @NotNull String... expectedActions) {
+        List<CodeAction> codeActions = JSONUtils.getLsp4jGson()
+                .fromJson(codeActionsJson, new TypeToken<List<CodeAction>>() {}.getType());
+
+        assertCodeActions(fileName, editorContentText, codeActions, expectedActions);
+    }
+
+    /**
+     * Test LSP code actions.
+     *
+     * @param fileName          the file name used to match registered language servers.
+     * @param editorContentText the editor content text.
+     * @param codeActions       the LSP CodeAction list.
+     * @param expectedActions   the expected IntelliJ intention action text.
+     */
+    protected void assertCodeActions(@NotNull String fileName,
+                                     @NotNull String editorContentText,
+                                     @NotNull List<CodeAction> codeActions,
+                                     @NotNull String... expectedActions) {
+        List<Either<Command, CodeAction>> wrappedCodeActions = codeActions.stream()
+                .distinct()
+                .map(Either::<Command, CodeAction>forRight)
+                .collect(Collectors.toList());
+
+        List<Diagnostic> diagnostics = codeActions.stream()
+                .flatMap(codeAction -> Optional.ofNullable(codeAction.getDiagnostics()).stream().flatMap(List::stream))
+                .distinct()
+                .collect(Collectors.toList());
+
+        MockLanguageServer.INSTANCE.setTimeToProceedQueries(200);
+        MockLanguageServer.INSTANCE.setCodeActions(wrappedCodeActions);
+        MockLanguageServer.INSTANCE.setDiagnostics(diagnostics);
+
+        myFixture.configureByText(fileName, editorContentText);
+        Collection<HighlightInfo> highlightInfos = myFixture.doHighlighting();
+
+        List<IntentionAction> actions = highlightInfos.stream()
+                .flatMap(info -> info.quickFixActionRanges != null
+                        ? info.quickFixActionRanges.stream().map(pair -> pair.getFirst().getAction())
+                        : Stream.empty())
+                .filter(Objects::nonNull)
+                .distinct()
+                .collect(Collectors.toList());
+
+        List<String> codeActionTitles = new ArrayList<>();
+        for (IntentionAction action : actions) {
+            if (action.isAvailable(myFixture.getProject(), myFixture.getEditor(), myFixture.getFile())) {
+                codeActionTitles.add(action.getText());
+            }
+        }
+        for (String expectedAction : expectedActions) {
+            assertContainsElements(codeActionTitles, expectedAction);
+        }
+    }
+
+    /**
+     * Test applying a code action.
+     */
+    public void assertApplyCodeAction(@NotNull String fileName,
```

Comment:
Change your signature like this:

 void assertApplyCodeAction(@NotNull String fileName,
                                      @NotNull String editorContentText,
                                      @NotNull String actionText,
                                      @NotNull String expectedEditorContentText,
                                     @NotNull CodeAction unresolved,
                                      @NotNull String resolvedCodeActionJson)